anyway/views/news_flash/api.py

```diff
@@ -281,3 +290,55 @@ def normalize_query_param(key, value: list):
 def normalize_query(params: dict):
     params_non_flat = params.to_dict(flat=False)
     return {k: normalize_query_param(k, v) for k, v in params_non_flat.items()}
+
+
+@roles_accepted(
+    BE_CONST.Roles2Names.Authenticated.value,
+    BE_CONST.Roles2Names.Location_verification.value,
+    need_all_permission=True,
+)
+def update_new_flash_qualifying(id):
+    current_user = get_current_user()
+    manual_update = False
+    use_road_segment = True
+
+    newsflash_location_qualification = request.values.get("newsflash_location_qualification")
+    road_segment_name = request.values.get("road_segment_name")
+    yishuv_name = request.values.get("yishuv_name")
+    street1_hebrew = request.values.get("street1_hebrew")
+    newsflash_location_qualification = QUALIFICATION_TO_ENUM_VALUE[newsflash_location_qualification]
+    if newsflash_location_qualification == NewsflashLocationQualification.MANUAL.value:
+        manual_update = True
+        if road_segment_name is None:
+            if (yishuv_name is None) or (street1_hebrew is None):
+                logging.error("manual update must include location detalis.")
+                return return_json_error(Es.BR_FIELD_MISSING)
+            else:
+                use_road_segment = False
+    else:
+        if road_segment_name is not None or street1_hebrew is not None or yishuv_name is not None:
+            logging.error("only manual update should contain location details.")
+            return return_json_error(Es.BR_BAD_FIELD)
+    news_flash_obj = db.session.query(NewsFlash).filter(NewsFlash.id == id).first()
+    if news_flash_obj is not None:
+        if manual_update:
+            if use_road_segment:
+                news_flash_obj.road_segment_name = road_segment_name
+                news_flash_obj.resolution = fields_to_resolution.get("road_segment_name")
+            else:
+                news_flash_obj.yishuv_name = yishuv_name
+                news_flash_obj.street1_hebrew = street1_hebrew
+                news_flash_obj.resolution = fields_to_resolution.get(
+                    ("yishuv_name", "street1_hebrew")
+                )
```

Comment:
in the current implemention yes, i build this tuple in the code.
if the location that the user set is yishuv_name and street1_hebrew it will be in this order:("yishuv_name", "street1_hebrew")